src/mpl/src/shm/mpl_shm_sysv.c

```diff
@@ -105,6 +114,38 @@ int MPL_shm_seg_attach(MPL_shm_hnd_t hnd, intptr_t seg_sz, void **shm_addr_ptr,
                                            MPLI_SHM_FLAG_SHM_ATTACH);
 }
 
+/* Create new SHM segment and attach to it with specified starting address
+ * hnd : A "init"ed shared mem handle
+ * seg_sz: Size of shared mem segment
+ * shm_addr_ptr (inout): Pointer to specified starting address, the address cannot be NULL.
+ *                       The actual attached memory address is updated at return.
+ * offset : Offset to attach the shared memory address to
+ */
+int MPL_shm_fixed_seg_create_and_attach(MPL_shm_hnd_t hnd, intptr_t seg_sz,
+                                        void **shm_addr_ptr, int offset)
+{
+    int rc = 0;
```

Comment:
Only comment is why not eliminate these `rc` variables and just return the function evaluation directly?

Reply:
You are right. It is unneeded. I will make the changes to all `MPL_shm_fixed_*` functions.

Reply:
Fixed. Would you mind if I merge it once the jenkins jobs are done ?

Reply:
No problem. Go ahead and merge 👍